subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -95,4 +91,31 @@ public DateTime GetNextPaymentDate(Subscription subscription)
         var nextPaymentDate = GetNextPaymentDate(subscription);
         return (nextPaymentDate <= _dateProvider.Today, nextPaymentDate);
     }
+
+    private static bool PaidThisMonth(Subscription subscription, DateTime monthDate) => subscription.LastPayment.Month == monthDate.Month && subscription.LastPayment.Year == monthDate.Year;
+
+    private static bool PaidInPast(Subscription subscription, DateTime monthDate) => (subscription.LastPayment.Month < monthDate.Month && subscription.LastPayment.Year == monthDate.Year) || subscription.LastPayment.Year < monthDate.Year;
+
+    private IEnumerable<Subscription> GetIterationsForMonth(Subscription subscription, DateTime monthDate)
+    {
+        var subscriptions = new List<Subscription>();
+        var subscriptionToUse = (Subscription)subscription.Clone();
+        while (PaidInPast(GetNextSubscription(subscriptionToUse), monthDate))
```

Comment:
Dont we already know its paid in past since we filtered away thoose that are not?

Reply:
we do. we're bringing that past subscription to the present with this loop. 
If a weekly subscription started in the first week of July (let's say the last payment is 7 Jul) and we're looking at it sometime around mid-August (14 Aug), we don't need the instances where next payment is in July while displaying subscriptions for August.